src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -189,6 +189,16 @@ void JavaThread::set_jvmti_vthread(oop p) {
   _jvmti_vthread.replace(p);
 }
 
+// If there is a virtual thread mounted then return vthread() oop.
+// Otherwise, return threadObj().
+oop JavaThread::vthread_or_thread() const {
+    oop result = vthread();
+    if (result == nullptr) {
```

Comment:
Thank you for the question.
This function does not care if this is a vthread, or not and if it is mounted or not.
We just need a right oop of currently executed thread or vthread.
The `threadOop()` and `vthread()` can return the same oop.
If `vthread()` is not `nullptr` then it can point either to a `j.l.Thread` or a `j.lVirtualThread`.
If it is `nullptr` then we have to take `threadOop()`.